### PR TITLE
CTCP-2362: Ensure we don't send a 500 if we don't find a box (return 404 instead)

### PR DESCRIPTION
**Pull request by drnaylor-oc**

### CODE REVIEW

**Review by sonisn1988 — DISMISSED** [review]
LGTM

File: app/uk/gov/hmrc/transitmovementspushnotifications/services/PushPullNotificationService.scala [hunk 1]
```diff
@@ -95,7 +99,7 @@ class PushPullNotificationServiceImpl @Inject() (pushPullNotificationConnector:
           case Right(_) => Right(())
           case Left(error) =>
             error.statusCode match {
-              case NOT_FOUND => Left(BoxNotFound(boxAssociation.boxId.value))
+              case NOT_FOUND => Left(BoxNotFound(boxAssociation.boxId))
```
Comment (liandra-giacomine) [inline app/uk/gov/hmrc/transitmovementspushnotifications/services/PushPullNotificationService.scala:102]: We're associating the same BoxNotFound error in checkBoxIdExists and sendPushNotification - but in sendPushNotification we have a boxId that did exist and no longer exists, just wonder if it's worth logging to differentiate between the two errors?

Comment (drnaylor-oc) [reply app/uk/gov/hmrc/transitmovementspushnotifications/services/PushPullNotificationService.scala:102]: That's actually an interesting point. In both cases, the error is effectively the same (there was no box), but the effect of that is different.

If a box has gone missing, that's not our fault, but there is a question as to whether we should be alerted. If the box has disappeared, the chances of that being our fault is zero, and the chances of it being PPNSs fault is slim at best, it's likely the trader has deleted a client managed box. For sure, we shouldn't alert on this. I'm kind of okay with the idea of logging though, perhaps just stating that a given box ID for a given movement no longer exists. Let me check if that's something that we think will be okay to log.

We _might_ want to consider an endpoint to allow a trader to update their associated box ID, but that's probably something that could come out of UR.

Comment (drnaylor-oc) [reply app/uk/gov/hmrc/transitmovementspushnotifications/services/PushPullNotificationService.scala:102]: We can log here -- client IDs and message IDs get logged elsewhere anyway, and could be linked by request ID. It's not a problem. I'll do that this afternoon.

Comment (liandra-giacomine) [reply app/uk/gov/hmrc/transitmovementspushnotifications/services/PushPullNotificationService.scala]: That's a good point about the endpoint to update the boxId -in the case where they don't subscribe to PPNS to begin with but then end up doing it later on in the journey?

**Review by liandra-giacomine — APPROVED** [review]

**Pull request merged**